cmd/producer/main_test.go

```diff
@@ -27,13 +27,55 @@ type fakeRedis struct {
 	client redis.Cmdable
 }
 
+func TestRedisClientSetup(t *testing.T) {
+	tests := []struct {
+		name			string
+		cert			string
+	}{{
+		name:			"test with fake cert from x509",
+		cert:			"-----BEGIN CERTIFICATE-----\n" +
+						"MIIDBjCCAe6gAwIBAgIRANXM5I3gjuqDfTp/PYrs+u8wDQYJKoZIhvcNAQELBQAw\n" +
+						"EjEQMA4GA1UEChMHQWNtZSBDbzAeFw0xODAzMjcxOTU2MjFaFw0xOTAzMjcxOTU2\n" +
```

Comment:
Why do we need to test with a “Fake” certificate?

Reply:
Test that the client works correctly both with and without a TLS certificate.  Bumps test coverage.  

This was documented as a supported feature (no TLS cert, local to the cluster redis) but was causing the errors I discovered in the async case when not using an empty TLS cert.